pkg/chunk/aws/dynamodb_storage_client.go

```diff
@@ -187,7 +141,7 @@ func (a dynamoDBStorageClient) NewWriteBatch() chunk.WriteBatch {
 	return dynamoDBWriteBatch(map[string][]*dynamodb.WriteRequest{})
 }
 
-func logWriteRetry(ctx context.Context, unprocessed dynamoDBWriteBatch) {
+func logWriteRetry(unprocessed dynamoDBWriteBatch, dynamoThrottled *prometheus.CounterVec) {
```

Comment:
You may consider to pass the entire `*metrics`. I would be less error prone, otherwise you may do a mistake and pass the wrong metric.